buildscripts/templates/SemanticAttributes.java.j2

```diff
@@ -719,7 +719,7 @@ public final class {{class}} {
   /**
    * Immediate client peer port number.
    *
-   * @deprecated This item has been renamed in 1.22.0 of the semantic conventions. Use {@link SemanticAttributes#NETWORK_PEER_PORT} on server telemetry and {@link SemanticAttributes#NETWORK_LOCAL_PORT} on client telemetry instead.
+   * @deprecated This item has been renamed in 1.22.0 of the semantic conventions. Use {@link SemanticAttributes#NETWORK_LOCAL_PORT} on server telemetry and {@link SemanticAttributes#NETWORK_PEER_PORT} on client telemetry instead.
    */
   @Deprecated
   public static final AttributeKey<Long> CLIENT_SOCKET_PORT = longKey("client.socket.port");
```

Comment:
I think the original is correct here? server telemetry means telemetry captured on the server side of a connection, maybe that's the confusion?

Reply:
The problem is that the JavaDoc of `SERVER_SOCKET_PORT` and `SERVER_SOCKET_ADDRESS` states the opposite for peer and local for client and server usage:
```java
  /**
   * @deprecated This item has been renamed in 1.22.0 of the semantic conventions. Use {@link
   *     SemanticAttributes#NETWORK_LOCAL_ADDRESS} on server telemetry and {@link
   *     SemanticAttributes#NETWORK_PEER_ADDRESS} on client telemetry instead.
   */
  @Deprecated
  public static final AttributeKey<String> SERVER_SOCKET_ADDRESS =
      stringKey("server.socket.address");
```
vs.
```java
  /**
   * @deprecated This item has been renamed in 1.22.0 of the semantic conventions. Use {@link
   *     SemanticAttributes#NETWORK_PEER_ADDRESS} on server telemetry and {@link
   *     SemanticAttributes#NETWORK_LOCAL_ADDRESS} on client telemetry instead.
   */
  @Deprecated
  public static final AttributeKey<String> CLIENT_SOCKET_ADDRESS =
      stringKey("client.socket.address");
```

So which association is correct?

```
peer <-> client
local <-> server
```

or

```
peer <-> server
local <-> client
```

Reply:
Or is this meant from the perspective of the instrumented application? So `local` should always be used for the current application side (either client or server) and `peer` for the remote application?

Sorry, but this is really confusing (at least just from the JavaDoc).

Reply:
> Or is this meant from the perspective of the instrumented application? So `local` should always be used for the current application side (either client or server) and `peer` for the remote application?

👍

> Sorry, but this is really confusing (at least just from the JavaDoc).

check out https://github.com/open-telemetry/semantic-conventions/blob/d21135e07ce9ca1ba05581021aebe84cc1d27ee8/model/registry/network.yaml#L112

and improvements (to both semconv and javadocs) are welcome!

Reply:
Thanks for the clarification! I'll see that I get a better wording for this PR by tomorrow, that matches the meaning of the spec.

Reply:
The PR is updated. For me the spec and description for the new attributes are fine.

So I've unified all 4 deprecated JavaDocs to the wording that matches the new attributes:
```
   * @deprecated This item has been renamed in 1.22.0 of the semantic conventions. Use {@link
   *     SemanticAttributes#NETWORK_LOCAL_ADDRESS} for local network telemetry and {@link
   *     SemanticAttributes#NETWORK_PEER_ADDRESS} for peer network telemetry instead.
   */
```